File: mobile/lib/core/auth/store.dart
class Store {
  Store({
    required this.id,
    required this.name,
    required this.code,
    required this.isMain,
    required this.isActive,
  });

  factory Store.fromJson(Map<String, dynamic> json) => Store(
    id: json['id'] as int,
    name: json['name'] as String,
    code: json['code'] as String,
    isMain: json['is_main'] as bool? ?? false,
    isActive: json['is_active'] as bool? ?? true,
  );

  final int id;
  final String name;
  final String code;
  final bool isMain;
  final bool isActive;
}

File: backend/app/Models/Store.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

#[Fillable(['name', 'code', 'is_main', 'address', 'phone', 'is_active'])]
class Store extends Model
{
    use HasFactory;

    protected function casts(): array
    {
        return [
            'is_main' => 'boolean',
            'is_active' => 'boolean',
        ];
    }

    public function users(): HasMany
    {
        return $this->hasMany(User::class);
    }

    public function stockMovements(): HasMany
    {
        return $this->hasMany(StockMovement::class);
    }

    public function itemStoreSettings(): HasMany
    {
        return $this->hasMany(ItemStoreSetting::class);
    }
}
